### PR TITLE
Feature/player

### PR DESCRIPTION
**Pull request by thabti**
cdrigcmqaw

### CODE REVIEW

**Comment by thabti** [comment]
screen shot 2017-07-10 at 1 21 16 am

**Comment by mmahalwy** [comment]
ROCKSTAR :fire::fire::fire::fire::fire::fire:

File: src/components/AudioPlayer/index.js [hunk 1]
```diff
@@ -249,6 +246,14 @@ class AudioPlayer extends Component {
             toggleVolume={this.toggleVolume}
           />
         </View>
+        <SongSlider
+          onSlidingStart={this.onSlidingStart}
```
Comment (mmahalwy) [inline src/components/AudioPlayer/index.js]: sorry, nit pick, but i think usually the handler function is named `handle...` and not `on...`. So `this.handleSlidingStart` :)

Comment (mmahalwy) [reply src/components/AudioPlayer/index.js]: Not a blocker though

File: src/container/AudioPlayer.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+const View = styled.View`
+  height: ${props => props.height}px;
+  width: 100%;
+  background: #000;
+  borderTopColor: #2ca4ab;
+  borderTopWidth: 5px;
+`;
+const AView = Animated.createAnimatedComponent(View);
```
Comment (mmahalwy) [inline src/container/AudioPlayer.js]: Nice!

File: src/container/AudioPlayer.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  align-self: flex-end;
+  align-items: center;
+  justify-content: center;
+  border-radius: 5px;
+  position: relative;
+  top: -3px;
+`;
+
+const Chevron = styled(FontAwesome)`
+ transform: rotate(${props => props.deg});
```
Comment (mmahalwy) [inline src/container/AudioPlayer.js]: Why not have this as a prop that either triggers up or down? That way if you forget to pass the prop. 

```
transform: ${props => props.directionUp ? 180 : 0}deg
```

File: src/container/AudioPlayer.js (hunk 2 above)
Comment (mmahalwy) [reply src/container/AudioPlayer.js]: Maybe name `AnimatedView` to be more descriptive?

File: src/container/AudioPlayer.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { Animated } from 'react-native';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as Actions from '../actions/index';
+import styled from 'styled-components/native';
+
+import AudioP from '../components/AudioPlayer';
```
Comment (mmahalwy) [inline src/container/AudioPlayer.js]: `Player`  ?

**Review by mmahalwy — APPROVED** [review]
Woot!

File: src/utils/perpareReciterSurahsList.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ export default (reciter, chapters, chapterId) => {
       id: chapters[index].id,
       path: `https://download.quranicaudio.com/quran/${reciter.relative_path}${zeroFill(3, chapters[index].id)}.mp3`,
       title: chapters[index].name.simple,
-      artist: reciter.name,
+      artist: reciter.name || 'hello',
```
Comment (AlaaAttyaMohamed) [inline src/utils/perpareReciterSurahsList.js]: hello? maybe we can have a more applicable default reciter title here. Maybe "Reciter"

Comment (AlaaAttyaMohamed) [reply src/utils/perpareReciterSurahsList.js]: btw we need to move all that static strings in the app to a Class that holds them as constants. We might also use this package in the future to support localization https://github.com/stefalda/ReactNativeLocalization

File: src/components/AudioPlayer/index.js [hunk 6]
```diff
@@ -196,36 +195,34 @@ class AudioPlayer extends Component {
   }
 
   render() {
-    if (this.state.songs < 1) {
-      return <Loader />;
-    }
+    // if (this.state.songs < 1) {
+    //   return <Loader />;
+    // }
```
Comment (AlaaAttyaMohamed) [inline src/components/AudioPlayer/index.js]: remove if it's not use anymore

File: src/components/PlayerButtons.js [hunk 7]
```diff
@@ -37,89 +36,66 @@ export const ForwardButton = (props) => {
...
-        style={Styles.downloadButton}
-        name="download"
-        size={25}
-        color="#333"
-      />
-    );
-  }
+export const SongSlider = (props) => {
+  const currentTime = Utils.secondTime(props.currentTime) || '00:00';
+  const durationTime = Utils.secondTime(props.songDuration) || '00:00';
```
Comment (AlaaAttyaMohamed) [inline src/components/PlayerButtons.js]: these static string ('00:00') might be moved to a class that holds such constants too

File: src/container/AudioPlayer.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+`;
+
+class AudioPlayer extends React.Component {
+  render() {
+    const { chapters, player, actions } = this.props;
+    const selection = chapters.selection;
+    if (selection && selection.chapter) {
+      return (
+        <AView height={player.minimise ? '50' : '200'}>
+          <ViewCloseIcon onPress={() => actions.minimisePlayer()}>
```
Comment (AlaaAttyaMohamed) [inline src/container/AudioPlayer.js]: and these default values too might be moved to a class that wraps such constants

File: src/container/Wrapper.js [hunk 9]
```diff
@@ -21,14 +20,18 @@ export class Wrapper extends Component {
...
     const { audio } = this.props;
 
     return (
       <View style={{ flex: 1 }}>
         {this.props.children}
-        <AudioPlayer data={{ chapters: audio.chapters, selected_chapter: audio.selected_chapter }} />
+        <AudioPlayer
+          data={{
+            chapters: audio.chapters,
+            selected_chapter: audio.selected_chapter
```
Comment (AlaaAttyaMohamed) [inline src/container/Wrapper.js:32]: we need to set a rule if we'll follow the underscore based or camel case based name convention for name variables

**Comment by AlaaAttyaMohamed** [comment]
@sabeurthabti i got surprised btw the design btw. Amazing job 👍

File: src/utils/perpareReciterSurahsList.js (hunk 5 above)
Comment (mmahalwy) [reply src/utils/perpareReciterSurahsList.js]: Good catch!

File: src/components/AudioPlayer/index.js (hunk 1 above)
Comment (thabti) [reply src/components/AudioPlayer/index.js]: agreed.

File: src/container/AudioPlayer.js (hunk 3 above)
Comment (thabti) [reply src/container/AudioPlayer.js]: pure laziness from my part. Didn't know what to call the prop. 

Will change.

File: src/utils/perpareReciterSurahsList.js (hunk 5 above)
Comment (thabti) [reply src/utils/perpareReciterSurahsList.js]: good catch, I only recently discovered hot reloading. I got tired of navigating the screens. I will remove this.

File: src/components/AudioPlayer/index.js (hunk 6 above)
Comment (thabti) [reply src/components/AudioPlayer/index.js]: I think I will put this back in.

File: src/container/AudioPlayer.js (hunk 8 above)
Comment (thabti) [reply src/container/AudioPlayer.js]: agreed, and better naming.

File: src/container/Wrapper.js (hunk 9 above)
Comment (thabti) [reply src/container/Wrapper.js:32]: I think the underscore comes from the API.

**Review by AlaaAttya — APPROVED** [review]

**Comment by AlaaAttya** [comment]
btw guys sorry for using two accounts for github the one i've used before was my working company one, this one i'm commenting from is my personal account. sorry for that confusion

**Comment by thabti** [comment]
merged, we will continue to improve the code base. @AlaaAttya @mmahalwy thanks for the review.

**Pull request merged**